services:
  auth:
    build: ./auth
    tty: true
    restart: always
    ports:
      - "8001:8000"
    env_file:
      - ./auth/.env
    healthcheck:
      test: [ "CMD", "curl", "-f", "http://auth:8000" ]
      interval: 30s
      timeout: 10s
      retries: 3
    depends_on:
      mongodb:
        condition: service_healthy
    networks:
      - backend

  mongodb:
    image: mongo:8.0.5
    restart: always
    ports:
      - "27018:27017"
    env_file:
      - ./auth/.env
    volumes:
      - mongo_data:/data/db
    healthcheck:
      test: [ "CMD", "mongosh", "--eval", "db.adminCommand('ping')" ]
      interval: 10s   # Проверять каждые 10 секунд
      timeout: 5s     # Ждать ответа 5 секунд
      retries: 5      # Попытаться 5 раз перед ошибкой
    networks:
      - backend

  nginx:
    image: nginx:latest
    restart: always
    volumes:
      - ./nginx/conf.d/default.conf:/etc/nginx/conf.d/default.conf
      - ./logs:/var/log/nginx
    ports:
      - "8080:80"
    depends_on:
      - auth
    healthcheck:
      test: [ "CMD", "curl", "-f", "http://nginx:80" ]
      interval: 30s
      timeout: 10s
      retries: 3
    networks:
      - backend

networks:
  backend:

volumes:
  mongo_data: